app/main.py

```diff
@@ -68,11 +70,24 @@
 # We define a few attributes of the app object
 app.title = 'Pyronear - Monitoring platform'
 app.config.suppress_callback_exceptions = True
-server = app.server   # Gunicorn will be looking for the server attribute of this module
-
-# We create a rough layout, filled with the content of the homepage
-app.layout = html.Div([dcc.Location(id='url', refresh=False),
-                       html.Div(id='page-content', children=Homepage())])
+server = app.server  # Gunicorn will be looking for the server attribute of this module
+
+# We create a rough layout, filled with the content of the homepage/alert page
+app.layout = html.Div(
+    [
+        dcc.Location(id="url", refresh=False),
+        html.Div(id="page-content", style={"height": "100%"}),
+        # Interval component to generate call to API every 10 seconds
+        dcc.Interval(
+            id="interval-component-homepage",
+            interval=10 * 1000
+        ),
+        # Hidden div to keep a record of live alerts data
```

Comment:
No, not really. According to the [documentation](https://dash.plotly.com/dash-core-components/store)

- The memory store reverts to the default on every page refresh
- The local store will take the initial data only the first time the page is loaded and keep it until it is cleared 
- The session store: same as the local store but will lose the data when the browser/tab closes 

Here I have the following ideas:
- for `div_live_alerts_data` : we get data from the API every 10 seconds. Therefore it's okay to loose it on refresh
- for `last_displayed_event_id`: cannot keep this variable in memory, it needs to be updated all the time as it is used in callbacks to know if we keep on displaying the same event or not
- for `images_url_current_alert`: I switched it to `session` because if the user refreshes the page (but let's hope he does not, don't think that Dash likes refresh), we will lose the urls. To avoid this, we need to store them into the browser.

Reply:
Thanks for the detailed info.

> * for `images_url_current_alert`: I switched it to `session` because if the user refreshes the page (but let's hope he does not, don't think that Dash likes refresh), we will lose the urls. To avoid this, we need to store them into the browser.

Agreed on the `session` storage.

Reply:
Very clear, thanks a lot!

It will probably be worth testing this cache mechanism in depth after we have implemented the webhook-based alert flow, the login, etc. To investigate what happens when the page is refreshed, when the app is opened in two tabs simultaneously, etc. So that we can identify the behaviours we want to keep and those that we may want to eliminate.